debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -40,16 +42,20 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(EventDispatcher.class);
 
-    private final TopicSelector topicSelector;
-    private final DatabaseSchema schema;
+    private final TopicSelector<T> topicSelector;
+    private final DatabaseSchema<T> schema;
+    private final HistorizedDatabaseSchema<T> historizedSchema;
     private final ChangeEventQueue<Object> queue;
     private final DataCollectionFilter<T> filter;
 
-    public EventDispatcher(TopicSelector topicSelector, DatabaseSchema schema,
+    public EventDispatcher(TopicSelector<T> topicSelector, DatabaseSchema<T> schema,
             ChangeEventQueue<Object> queue,
             DataCollectionFilter<T> filter) {
         this.topicSelector = topicSelector;
         this.schema = schema;
+        this.historizedSchema = schema instanceof HistorizedRelationalDatabaseSchema
```

Comment:
Is `HistorizedRelationalDatabaseSchema` really subtype of `HistorizedDatabaseSchema`?

Reply:
Could catch, I should check for HistorizedDatabaseSchema here, as this class shouldn't care wether relational or not. That said, indeed HistorizedRelationalDatabaseSchema should extend HistorizedDatabaseSchema. I'll change the two things and merge.